src/libraries/icubmod/embObjLib/ethBoards.cpp

```diff
@@ -54,15 +56,31 @@
 
 eth::EthBoards::EthBoards()
 {
-    memset(LUT, 0, sizeof(LUT));
+    // memset(LUT, 0, sizeof(LUT));
+    // Changed memset with for loop initialization due to problem on Windows
+    // which was failing in updating correctly values to LUT vars
+    for (auto& lut : LUT)
+    {
+        lut.ipv4 = 0;
+        lut.name = "";
+        lut.numberofinterfaces = 0;
+        lut.boardnumber = 0;
+        lut.resource = nullptr;
+        for (auto& itf : lut.interfaces) {itf = nullptr;}
+    }
+    
     sizeofLUT = 0;
 }
 
 
 eth::EthBoards::~EthBoards()
 {
-    memset(LUT, 0, sizeof(LUT));
-    sizeofLUT = 0;
+    /**
+    * LUT has already been cleaned by the call to
+    * eth::EthBoards::rem(eth::AbstractEthResource* res, iethresType_t type) and
+    * eth::EthBoards::rem(eth::AbstractEthResource* res)
+    * so there's nothing to do here
+    */
```

Comment:
Are the method `rem` called also while deleting an object of this class?

Reply:
OK, what happens if we call the dtor w/out calling every `rem()` ? 
Probably nothing because LUT is destroyed by its default dtor that works OK. 

In any case, if you want you can keep the clear()

```C++
eth::EthBoards::~EthBoards()
{
    // nothing is required because LUT is destroyed correctly
    // BUT i just do the following anyway:
   for (auto& l : LUT)
   { 
       l.clear();
   }
    sizeofLUT = 0;
}
```

Reply:
Exactly, as you say @marcoaccame nothing happens, since LUT has been already destroyed by the default dtor when calling `rem()`